#!/usr/bin/env python
PACKAGE = "control_sys"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("roll_p",    double_t,    0, "roll controller P gain", 1,  0, 2)
gen.add("roll_i",    double_t,    0, "roll controller I gain", 1,  0, 2)

gen.add("pitch_p",    double_t,    0, "pitch controller P gain", 1,  0, 2)
gen.add("pitch_i",    double_t,    0, "pitch controller I gain", 1,  0, 2)

gen.add("yaw_p",    double_t,    0, "yaw controller P gain", 0,  0, 100)
gen.add("yaw_i",    double_t,    0, "yaw controller I gain", 0,  0, 100)

gen.add("z_p",    double_t,    0, "z controller P gain", 2,  0, 2)
gen.add("z_i",    double_t,    0, "z controller I gain", 1,  0, 2)

gen.add("roll_cmd",    int_t,    0, "roll command", 0,  -1, 1)
gen.add("pitch_cmd",    int_t,    0, "pitch command", 0,  -1, 1)
gen.add("yaw_cmd",    int_t,    0, "yaw command",  0, -180, 180)

gen.add("altitude_cmd",    int_t,    0, "z command", 15, 15, 100)

gen.add("reset", bool_t, 0, "When activated, error integrals = 0. While active, error integration stops and all angle command = 0", True);

exit(gen.generate(PACKAGE, "control_sys", "angle_param"))
